cuqi/testproblem/_testproblem.py

```diff
@@ -148,69 +152,109 @@ class Deconvolution1D(BayesianProblem):
-        'hat' - a triangular hat function
-        'bumps' - two bumps
-        'derivGauss' - the first derivative of Gauss function
-        'pc' - Piece-wise constant phantom
-        ndarray - a custom phantom
+        | The phantom that is sampled to produce the exact solution (signal).
+        | 'Gauss' - a Gaussian function
+        | 'sinc' - a sinc function
+        | 'vonMises' - a periodic version of the Gauss function
+        | 'square' - a "top hat" function
+        | 'hat' - a triangular hat function
+        | 'bumps' - two bumps
+        | 'derivGauss' - the first derivative of Gauss function
+        | 'pc' - Piece-wise constant phantom
+        | ndarray - a custom phantom
 
     phantom_param : scalar
-        A parameter that determines the width of the central 
-        "bump" of the function; the larger the parameter,
-        the narrower the "bump."  
-        Does not apply to phantom = 'bumps' or ndarray.
+        | A parameter that determines the width of the central 
+        | peak of the function; the larger the parameter,
+        | the narrower the peak.  
+        | Does not apply to phantom = 'bumps' or ndarray.
 
     noise_type : string
-        The type of noise
-        "Gaussian" - Gaussian white noise¨
-        "scaledGaussian" - Scaled (by data) Gaussian noise
+        | The type of noise
+        | "Gaussian" - Gaussian white noise¨
+        | "scaledGaussian" - Scaled (by data) Gaussian noise. Here noise is scaled
+        | by each data point individually.
 
     noise_std : scalar
         Standard deviation of the noise
 
     prior : cuqi.distribution.Distribution, Default Gaussian
         Distribution of the prior
 
+    use_legacy : bool, Default False
+        If True, use the legacy matrix representation of the forward model.
+        This uses more memory but is faster for small problems.
+        
+
     """
     def __init__(self,
         dim=128,
-        kernel="gauss",
-        kernel_param=None,
-        phantom="gauss",
+        PSF="gauss",
+        PSF_param=None,
+        PSF_size=None,
+        BC="periodic",
+        phantom="sinc",
         phantom_param=None,
         noise_type="gaussian",
-        noise_std=0.05,
+        noise_std=0.01,
         prior=None,
-        data=None,
+        use_legacy=False,
         ):
         
+        # Set up forward model
+        if use_legacy: # Legacy matrix representation
+            if BC != "periodic":
+                raise ValueError("Legacy matrix representation only supports periodic boundary conditions")
+            A = _getCirculantMatrix(dim, PSF, PSF_param)
+            model = cuqi.model.LinearModel(A, range_geometry=Continuous1D(dim), domain_geometry=Continuous1D(dim))
+        else:
+            Afun = _getConvolutionOperator(dim, PSF, PSF_param, PSF_size, BC)
+
+            # Create matrix from Afun
+            Id = np.eye(dim)
+            A = np.array([Afun(Id[:, i]) for i in range(dim)])
```

Comment:
Seems this would be very slow, explicitly constructing the complete matrix, why is it we need it, couldn't we convolve with the PSF from which Afun is constructed?

Reply:
We construct the matrix explicitly at creation for this 1D case so subsequent calls are faster. This is fine since the matrix can fit in memory for 1D cases.